src/Compiler/CompilerOpenFPGA.cpp

```diff
@@ -492,6 +492,20 @@ bool CompilerOpenFPGA::RegisterCommands(TclInterpreter* interp,
   };
   interp->registerCmd("flat_routing", flat_routing, this, 0);
 
+  auto use_vpr_latest = [](void* clientData, Tcl_Interp* interp, int argc,
+                           const char* argv[]) -> int {
+    CompilerOpenFPGA& comp = *(CompilerOpenFPGA*)clientData;
+    std::filesystem::path vpr = comp.m_vprExecutablePath.lexically_normal();
+    try {
+      vpr.replace_filename("vpr_latest");
+      comp.VprExecPath(vpr);
+    } catch (...) {
+      assert(0);
```

Comment:
Asserts are easy for developers to debug.
They are a nightmare for end-users.
Please use the ErrorMessage instead and print a proper message like could not find this executable and return Tcl Error, so the Tcl script stops accordingly with the proper error trace.

Reply:
@alaindargelas assert() is enabled only in 'make debug' so the end users won't experience it.
replace_filename is a purely lexical (string) operation, it does not care if the paths exist.
I don't know what kind of exceptions it can throw, but cppreference and other references say it can, potentially.

replace_extension() is already used in other places in FOEDAG without catching exceptions, like some other C++ utilities that may throw are used without checking exceptions in FOEDAG. This assert(0) and exception is extremely unlikely case, I'll remove the try-catch to be consistent with FOEDAG code :).